pkg/controller/projects/projects/project.go

```diff
@@ -150,6 +159,57 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errNotProject)
 	}
 
+	keySecretRef := cr.Spec.ForProvider.ImportUrlSecretRef
+	importUrl := cr.Spec.ForProvider.ImportURL
+
+	// User has provided secret for the import
+	secretRefIsNotEmpty := keySecretRef != nil && keySecretRef.Namespace != "" && keySecretRef.Name != ""
+	keysAreNotEmpty := keySecretRef != nil && keySecretRef.PasswordKey != "" && keySecretRef.UsernameKey != ""
+	importUrlIsNotEmpty := importUrl != nil && *importUrl != ""
+
+	if secretRefIsNotEmpty && keysAreNotEmpty && importUrlIsNotEmpty {
+		// Retrieve secret from k8s
+		namespacedName := types.NamespacedName{
+			Namespace: keySecretRef.Namespace,
+			Name:      keySecretRef.Name,
+		}
+
+		secret := &corev1.Secret{}
+
+		err := e.kube.Get(ctx, namespacedName, secret)
+
+		if err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errNoImportSecret)
+		}
+
+		// Obtain the password from the secret.
+		password := string(secret.Data[keySecretRef.PasswordKey])
+		if password == "" {
+			return managed.ExternalCreation{}, errors.Wrap(err, errPasswordNotFound)
+		}
+
+		// Obtain the username from the secret.
+		username := string(secret.Data[keySecretRef.UsernameKey])
+		if username == "" {
+			return managed.ExternalCreation{}, errors.Wrap(err, errUsernameNotFound)
+		}
+
+		// manipulate url to add the secret. If secret is already in the url, it should be overridden
+		// https://username:password@example.com
+		parsedUrl, err := url.Parse(*importUrl)
+
+		if err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errParseUrlFailed)
+		}
+
+		userInfo := url.UserPassword(username, password)
+
+		parsedUrl.User = userInfo
+
+		// Override importUrl with the manipulated URL containing the credentials found in ImportSecretRef.
+		*importUrl = parsedUrl.String()
```

Comment:
This whole code segment is put into `Create` which is the wrong place for that.

It should be put into `LateInitialize` which runs during `Observed`and should only update `spec.forProvider.importUrl` if it is nil. 

`Create` itself should only rely on `spec.forProvider.importUrl` and not `importUrlSecretRef`.